testing/ase/test_enum_c_ase.cpp

```diff
@@ -0,0 +1,380 @@
+    fpga_token tok;
+};
+
+/**
+ * @test       nullfilter
+ *
+ * @brief      When the fpga_properties * parameter to fpgaEnumerate is nullptr, the
+ *             function returns FPGA_INVALID_PARAM.
+ */
+TEST_F(enum_c_ase_p, nullfilter) {
+  uint32_t matches = 0;
+  EXPECT_EQ(fpgaEnumerate(nullptr, 1, &tok, 1, &matches),
+            FPGA_INVALID_PARAM);
+}
+
+/**
+ * @test       nullmatches
+ *
+ * @brief      When the uint32_t *num_matches parameter to fpgaEnumerate is nullptr, the
+ *             function returns FPGA_INVALID_PARAM.
+ */
+TEST_F(enum_c_ase_p, nullmatches) {
+  uint32_t matches = 0;
+  EXPECT_EQ(fpgaEnumerate(&filter_, 1, &tok, 1, NULL),
+            FPGA_INVALID_PARAM);
+  EXPECT_EQ(
+      fpgaEnumerate(&filter_, 0, &tok, 1, &matches),
+      FPGA_INVALID_PARAM);
+}
+
+/**
+ * @test       nulltokens
+ *
+ * @brief      When the fpga_token* parameter to fpgaEnumerate is nullptr, the
+ *             function returns FPGA_INVALID_PARAM.
+ *
+ */
+TEST_F(enum_c_ase_p, nulltokens) {
+  uint32_t matches = 0;
+  EXPECT_EQ(fpgaEnumerate(&filter_, 0, NULL, 1, &matches),
+            FPGA_INVALID_PARAM);
+}
+
+/**
+ * @test       api_guid_to_fpga
+ *
+ * @brief      Test internal function api_guid_to_fpga in enum.c,
+ *             api_guid_to_fpga returns the expected guid.
+ *
+ */
+TEST(enum_c_ase, api_guid_to_fpga) {
+    fpga_guid guid;
+    uint8_t *guid_;
+    uint64_t guidh = 0x4041424344454647;
+    uint64_t guidl = 0x3031323334353637;;
+    api_guid_to_fpga(guidh, guidl, guid);
+    guid_ = (uint8_t*)guid;
+    for (int i=0; i<8; i++)
+        EXPECT_EQ(guid_[i], (0x40 + i));
+    for (int i=0; i<8; i++)
+        EXPECT_EQ(guid_[i+8], (0x30 + i));
+}
+
+/**
+ * @test       matches_filters_1
+ *
+ * @brief      Test internal function matches_filters(), the function returns
+ *             true.
+ */
+TEST_F(enum_c_ase_p, matches_filters_1) {
+  uint64_t j = 0;
+  struct _fpga_properties* _prop = (struct _fpga_properties*)filter_;
+
+  // Set the property valid field
+  SET_FIELD_VALID(_prop, FPGA_PROPERTY_FUNCTION);
+  _prop->function = 0xAE;
+
+  EXPECT_EQ(matches_filters(&filter_, 1, &tok,  &j), true);  
+}
+
+/**
+ * @test       matches_filters_2
+ *
+ * @brief      Test internal function matches_filters(), the function returns
+ *             true.
+ */
+TEST_F(enum_c_ase_p, matches_filters_2) {
+  uint64_t j = 0;
+  struct _fpga_properties* _prop = (struct _fpga_properties*)filter_;
+
+  SET_FIELD_VALID(_prop, FPGA_PROPERTY_DEVICE);
+  _prop->device = 0xAE;
+  
+  EXPECT_EQ(matches_filters(&filter_, 1, &tok,  &j), true);  
+}
+
+/**
+ * @test       matches_filters_3
+ *
+ * @brief      Test internal function matches_filters(), the function returns
+ *             true.
+ */
+TEST_F(enum_c_ase_p, matches_filters_3) {
+  uint64_t j = 0;
+  struct _fpga_properties* _prop = (struct _fpga_properties*)filter_;
+
+  _prop->valid_fields = 0;
+  EXPECT_EQ(matches_filters(&filter_, 1, &tok,  &j), true);
```

Comment:
I think you're testing the case where the given filter is empty, matching all devices.  Should update the comments..

Reply:
sure.